components/script/body.rs

```diff
@@ -166,6 +176,19 @@ fn run_form_data_algorithm(root: &GlobalScope, bytes: Vec<u8>, mime: &[u8]) -> F
     }
 }
 
+#[allow(unsafe_code)]
+fn run_array_buffer_data_algorithm(cx: *mut JSContext, bytes: Vec<u8>) -> Fallible<FetchedData> {
+    rooted!(in(cx) let mut array_buffer_ptr = ptr::null_mut::<JSObject>());
+    unsafe {
+        let arraybuffer = ArrayBuffer::create(cx, CreateWith::Slice(&bytes), array_buffer_ptr.handle_mut());
+        if arraybuffer.is_err() {
+            return Err(Error::JSFailed);
+        }
+        let rooted_heap = RootedTraceableBox::from_box(Heap::boxed(jsval::ObjectValue(array_buffer_ptr.get())));
```

Comment:
appreciate for suggestion and tried to update code as suggested, except creating boxed heap using ObjectValue - since type of ArrayBuffer is `RootedTraceableBox<Heap<JSVal>>`, using ptr directly creates `RootedTraceableBox<Heap<mut *JSObject>>`.

Reply:
Oh, let's make it a bit cleaner then - store `RootedTraceableBox<Heap<*mut JSObject>>` and get rid of the `jsval::ObjectValue`.